### PR TITLE
feat(security): org-wide mandatory two-factor enrolment (ADR-018)

### PR DESCRIPTION
**Pull request by CybotTM**
## What

Adds an **opt-in, org-wide mandatory 2FA** mode. When `REQUIRE_TWO_FACTOR=true`, every user must enrol a second factor — an authenticator app **or** a passkey — before they can use the app. Default is **off**, so nothing changes unless an operator opts in.

This completes the trio of 2FA governance features (alongside #524 self-service re-auth and #525 admin reset).

## Backend — the security boundary

- **`TwoFactorStatusService::hasTwoFactor()`** — a user has a second factor if TOTP is enrolled **or** they have ≥1 passkey (possession + user verification). Backed by a new `WebauthnCredentialRepository::countByUserHandle()`.
- **`RequireTwoFactorSubscriber`** (`kernel.request`, priority 7 — after the firewall) is **fail-closed**: an authenticated user without a second factor may reach only an allowlist — the SPA shell (which hosts the gate), the enrolment endpoints (`/settings/2fa/*`, `/settings/security/passkeys/*`), the login/logout ceremony, and static assets. Everything else is refused: **403 JSON** for API/XHR, **redirect to the SPA shell** for an HTML navigation. A login still resolving its 2FA challenge (`IS_AUTHENTICATED_2FA_IN_PROGRESS`) is exempt.
- The flag is an env-backed parameter (`app_require_two_factor` ← `REQUIRE_TWO_FACTOR`, default `false`), mirroring the existing `APP_SHOW_BILLABLE_FIELD_IN_EXPORT` pattern. `SpaAction` publishes the flag + the user's current 2FA state to the SPA.

## Frontend — the matching UX

- A full-screen **`TwoFactorGate`** renders in place of the app (in `main.tsx`, before the router) when 2FA is required and the user has none. It **reuses** the Settings enrolment controls (`TwoFactorControls`/`PasskeyControls`, now exported with optional completion callbacks); finishing either flow reloads into the app. A logout link is the escape hatch.
- New `twoFactorRequired` / `hasTwoFactor` config keys + a `needsTwoFactorEnrolment()` helper. Gate copy across all five catalogs (en/de/es/fr/ru).

## Tests

- **`TwoFactorStatusServiceTest`** (4): TOTP short-circuits the passkey query; a passkey counts; no-TOTP-no-handle skips the query; a handle with zero passkeys is not a factor.
- **`RequireTwoFactorSubscriberTest`** (17): flag off / unauthenticated / 2FA-in-progress / already-has-factor / sub-request → pass through; the enrolment+shell+ceremony allowlist stays reachable; `/getAllProjects` and `/settings/save` → 403; HTML navigation → redirect to `/ui/`.
- **Frontend** (`TwoFactorGate.test.tsx`, `config.test.ts`): the gate explains the requirement and offers enrolment + logout; completing TOTP or registering a passkey reloads; `needsTwoFactorEnrolment()` truth table.
- Full backend suite green except pre-existing `LdapClientServiceIntegrationTest` (needs a live LDAP server, unrelated); full frontend suite 379/379.

Local gates: PHPStan level 10 clean, phpat architecture clean, cs-fixer clean, frontend typecheck + lint clean.

## Deferred

A **runtime admin-UI toggle** is intentionally out of scope — there is no runtime settings store yet, so mandatory 2FA is an operator/env decision for now. When a settings store lands, the toggle can drive the same `app_require_two_factor` path.

### CODE REVIEW

File: src/EventSubscriber/RequireTwoFactorSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        '/logout',
+        '/2fa',                         // /2fa, /2fa_check (login challenge)
+        '/settings/2fa',                // TOTP start / confirm / disable
+        '/settings/security/passkeys',  // passkey register / list / delete
+        '/status',                      // health checks
+        '/build-ui',                    // Vite-built assets
+        '/css',
+        '/js',
+        '/images',
+    ];
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply src/EventSubscriber/RequireTwoFactorSubscriber.php:69]: Adopted — good catch for a fail-closed gate. `isAllowlisted` now uses a **strict boundary** (`$path === $prefix || str_starts_with($path, $prefix . '/')`) so a route like `/settings/2fa_bypass` or `/status_admin` can no longer ride in on a shared prefix, and the underscore ceremony paths `/login_check` and `/2fa_check` are now listed explicitly. Added regression tests for the substring-boundary cases (`/settings/2fa_bypass`, `/status_admin`, `/uixyz`, `/logout-all` → all 403). Also satisfies the Rector `array_any` suggestion. Fixed in the latest push.

File: src/EventSubscriber/RequireTwoFactorSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    private function isAllowlisted(string $path): bool
+    {
+        foreach (self::ALLOWLIST_PREFIXES as $prefix) {
+            if (str_starts_with($path, $prefix)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply src/EventSubscriber/RequireTwoFactorSubscriber.php:136]: Adopted — good catch for a fail-closed gate. `isAllowlisted` now uses a **strict boundary** (`$path === $prefix || str_starts_with($path, $prefix . '/')`) so a route like `/settings/2fa_bypass` or `/status_admin` can no longer ride in on a shared prefix, and the underscore ceremony paths `/login_check` and `/2fa_check` are now listed explicitly. Added regression tests for the substring-boundary cases (`/settings/2fa_bypass`, `/status_admin`, `/uixyz`, `/logout-all` → all 403). Also satisfies the Rector `array_any` suggestion. Fixed in the latest push.
